spark-indexer/src/main/java/org/uniprot/store/spark/indexer/chebi/ChebiRDDReader.java

```diff
@@ -1,35 +1,39 @@
 package org.uniprot.store.spark.indexer.chebi;
 
-import static org.uniprot.store.spark.indexer.util.SparkUtils.getInputReleaseDirPath;
+import static org.uniprot.store.spark.indexer.common.util.SparkUtils.getInputReleaseDirPath;
 
 import java.util.ResourceBundle;
 
 import org.apache.spark.api.java.JavaPairRDD;
-import org.apache.spark.api.java.JavaSparkContext;
-import org.apache.spark.sql.SparkSession;
 import org.uniprot.core.cv.chebi.ChebiEntry;
+import org.uniprot.store.spark.indexer.common.JobParameter;
 
 /**
  * @author lgonzales
  * @since 2020-01-17
  */
 public class ChebiRDDReader {
 
+    private ChebiRDDReader() {}
+
     /** @return JavaPairRDD{key=chebiId, value={@link ChebiEntry}} */
-    public static JavaPairRDD<String, ChebiEntry> load(
-            JavaSparkContext jsc, ResourceBundle applicationConfig, String releaseName) {
-        String releaseInputDir = getInputReleaseDirPath(applicationConfig, releaseName);
-        String filePath = releaseInputDir + applicationConfig.getString("chebi.file.path");
-
-        jsc.hadoopConfiguration().set("textinputformat.record.delimiter", "\n\n");
-        SparkSession spark = SparkSession.builder().config(jsc.getConf()).getOrCreate();
-
-        return (JavaPairRDD<String, ChebiEntry>)
-                jsc.textFile(filePath)
-                        .filter(
-                                input ->
-                                        !input.startsWith("format-version")
-                                                && !input.startsWith("[Typedef]"))
-                        .mapToPair(new ChebiFileMapper());
+    public static JavaPairRDD<String, ChebiEntry> load(JobParameter jobParameter) {
+        ResourceBundle config = jobParameter.getApplicationConfig();
+        String releaseInputDir = getInputReleaseDirPath(config, jobParameter.getReleaseName());
+        String filePath = releaseInputDir + config.getString("chebi.file.path");
+
+        jobParameter
+                .getSparkContext()
+                .hadoopConfiguration()
+                .set("textinputformat.record.delimiter", "\n\n");
+
+        return jobParameter
+                .getSparkContext()
+                .textFile(filePath)
+                .filter(
+                        input ->
+                                !input.startsWith("format-version")
+                                        && !input.startsWith("[Typedef]"))
+                .mapToPair(new ChebiFileMapper());
     }
```

Comment:
Wrap the conditions  !input.startsWith("format-version")
                                        && !input.startsWith("[Typedef]") into a named method like !isMeta(input) or isDataRecord(input)